addons/website_event/controllers/main.py

```diff
@@ -338,7 +458,15 @@ def _create_attendees_from_registration_post(self, event, registration_data):
         visitor_sudo = request.env['website.visitor']._get_visitor_from_request(force_create=True)
 
         registrations_to_create = []
+        updated_registrations = request.env['event.registration'].sudo()
         for registration_values in registration_data:
+            # For already existing registration, just write de new values to the record
+            if 'registration_id' in registration_values:
+                registration = request.env['event.registration'].sudo().browse(registration_values['registration_id'])
+                del registration_values['registration_id']
+                registration.write(registration_values)
+                updated_registrations |= registration
```

Comment:
```suggestion
                # registration = request.env['event.registration'].sudo().browse(registration_values['registration_id'])
                # del registration_values['registration_id']
                # registration.write(registration_values)
                # updated_registrations |= registration
```
If the user is public, this code will be executed but at the env.flush_all() it will say :
```
You are not allowed to access 'Event Registration' (event.registration) records.
This operation is allowed for the following groups:
- Events/Administrator
- Events/Registration Desk
```
But if I just comment the code, it works fine... As I already use sudo(), I don't know what cause this issue... :thinking:

Reply:
Réponse de RETH sur discord : https://discord.com/channels/678381219515465750/1255872827664498811/1256117445668634667
> Stephane m'a repondu en DM que c'etait ok de faire un fix locale parceque c'etait un probleme connu, donc oui il faut flush le recordset manuellement avec sudo pour le moment apparement :/
> 
> pour reference https://github.com/odoo/odoo/blob/17.0/odoo/fields.py#L3559 ici on "set defaults" mais il faut avoir acces au valeurs existantes pour ne pas overwrite, du coup on fait un read dans le flush mais sans etre sure qu'on avait le droit -> si properties doit etre recompute et que t'as pas acces au modele a la base, genre dans une route publique, tu dois flush manuellement avec sudo
> 
> Alors que je pense que globalement on suppose que quand on flush on a plus besoin du sudo, puisque les valeures sont deja en cache ou appliquees dans la transaction. Le fait que le champ ai besoin de droit d'acces pour fonctionner lors de la phase de compute n'est pas courant